pkg/virt-operator/resource/apply/instancetypes.go

```diff
@@ -0,0 +1,134 @@
+package apply
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	v1 "kubevirt.io/api/core/v1"
+	instancetypev1beta1 "kubevirt.io/api/instancetype/v1beta1"
+	"kubevirt.io/client-go/log"
+)
+
+func (r *Reconciler) createOrUpdateInstancetypes() error {
+	for _, instancetype := range r.targetStrategy.Instancetypes() {
+		if err := r.createOrUpdateInstancetype(instancetype.DeepCopy()); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *Reconciler) createOrUpdateInstancetype(instancetype *instancetypev1beta1.VirtualMachineClusterInstancetype) error {
+	instancetypeClient := r.clientset.VirtualMachineClusterInstancetype()
+
+	foundObj, err := instancetypeClient.Get(context.Background(), instancetype.Name, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
```

Comment:
@alicefr 
Here we need to use cache instead otherwise on each cycle this takes a lot of time.